src/js/actions/search/menucommands.js

```diff
@@ -0,0 +1,176 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var _ = require("lodash"),
+        Immutable = require("immutable"),
+        OS = require("adapter/os"),
+        keyUtil = require("js/util/key"),
+        strings = require("i18n!nls/strings");
+
+    var events = require("js/events");
+
+    var menuActions = require("js/actions/menu");
+
+    /**
+     * Get a localized label for the given menu entry ID
+     *
+     * @private
+     * @param {string} id
+     * @return {string|Object.<string, string>}
+     */
+    var _getLabelForEntry = function (id) {
+        var parts = id.split("."),
+            labels = strings.MENU,
+            path = "";
+
+        parts.forEach(function (part) {
+            if (labels[part] === undefined) {
+                path = null;
+                return false;
+            }
+
+            if (labels[part].$MENU) {
+                path += labels[part].$MENU + ">";
+                labels = labels[part];
+            } else {
+                path += labels[part];
+            }
+        });
+
+        return path;
+    };
+
+    var _getShortcut = function (fullShortcut) {
+        var modifierBits = fullShortcut.modifiers,
+            keyChar = fullShortcut.keyChar,
+            keyCode = fullShortcut.keyCode,
+            modifierStrings = strings.SEARCH.MODIFIERS,
+            shortcut = "";
+
+        if (modifierBits) {
+            var modifiers = keyUtil.bitsToModifiers(modifierBits);
+            
+            _.forEach(Object.keys(modifiers), function (key) {
+                if (modifiers[key]) {
+                    shortcut += modifierStrings[key.toUpperCase()] + "+";
+                }
+            });
+        }
+
+        if (keyChar) {
+            shortcut += keyChar.toString().toUpperCase();
+        }
+        if (keyCode) {
+            // Currently just have keys for F12 and escape
```

Comment:
Don't leave this to the next person. You can find all the eventKeyCodes here, and this file is perfect for a lookup table: https://github.com/adobe-photoshop/spaces-adapter/blob/master/test/spec/low-level-test.js#L2049